expression/builtin_other_vec_test.go

```diff
@@ -29,6 +40,87 @@ var vecBuiltinOtherCases = map[string][]vecExprBenchCase{
 	},
 	ast.BitCount: {},
+	ast.In: {
+		{
+			retEvalType: types.ETInt,
+			childrenTypes: []types.EvalType{
+				types.ETInt,
+				types.ETInt, types.ETInt, types.ETInt, types.ETInt,
+				types.ETInt,
+			},
+			constants: []*Constant{
+				nil,
+				nil, nil, nil, nil,
+				{Value: types.NewDatum(1), RetType: types.NewFieldType(mysql.TypeInt24)},
+			},
+			geners: []dataGenerator{&rangeInt64Gener{1, 2}, nil, nil, nil, nil},
+		},
+		{
+			retEvalType: types.ETInt,
+			childrenTypes: []types.EvalType{
+				types.ETString,
+				types.ETString, types.ETString, types.ETString, types.ETString,
+				types.ETString, //types.ETString, types.ETString, types.ETString,
+			},
+			constants: []*Constant{
+				nil,
+				nil, nil, nil, nil,
+				{Value: types.NewStringDatum("aaaaaaaaaa"), RetType: types.NewFieldType(mysql.TypeString)},
+				//{Value: types.NewDatum("bbbbbbbbbb"), RetType: types.NewFieldType(mysql.TypeString)},
+				//{Value: types.NewDatum("cccccccccc"), RetType: types.NewFieldType(mysql.TypeString)},
+				//{Value: types.NewDatum("dddddddddd"), RetType: types.NewFieldType(mysql.TypeString)},
+			},
+			geners: []dataGenerator{&constStrGener{"aaaaaaaaaa"}, nil, nil, nil, nil},
+		},
+		{
+			retEvalType: types.ETInt,
+			childrenTypes: []types.EvalType{
+				types.ETDatetime,
+				types.ETDatetime, types.ETDatetime,
+			},
+			constants: []*Constant{
+				nil,
+				{Value: types.NewTimeDatum(dateTimeFromString("2019-01-01")), RetType: types.NewFieldType(mysql.TypeDatetime)},
+				{Value: types.NewTimeDatum(dateTimeFromString("2019-01-01")), RetType: types.NewFieldType(mysql.TypeDatetime)},
+			},
+		},
+		{
+			retEvalType: types.ETInt,
+			childrenTypes: []types.EvalType{
+				types.ETJson,
+				types.ETJson, //types.ETJson,
+			},
+			constants: []*Constant{
+				nil,
+				{Value: types.NewJSONDatum(json.CreateBinary("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")), RetType: types.NewFieldType(mysql.TypeJSON)},
+				//{Value: types.NewJSONDatum(json.CreateBinary("bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb")), RetType: types.NewFieldType(mysql.TypeJSON)},
+			},
+		},
+		{
+			retEvalType: types.ETInt,
+			childrenTypes: []types.EvalType{
+				types.ETDuration,
+				types.ETDuration, types.ETDuration,
+			},
+			constants: []*Constant{
+				nil,
+				{Value: types.NewDurationDatum(types.Duration{Duration: time.Duration(1000)}), RetType: types.NewFieldType(mysql.TypeDuration)},
+				{Value: types.NewDurationDatum(types.Duration{Duration: time.Duration(2000)}), RetType: types.NewFieldType(mysql.TypeDuration)},
+			},
+		},
+		{
+			retEvalType: types.ETInt,
+			childrenTypes: []types.EvalType{
+				types.ETReal,
+				types.ETReal, types.ETReal,
+			},
+			constants: []*Constant{
+				nil,
+				{Value: types.NewFloat64Datum(0.1), RetType: types.NewFieldType(mysql.TypeFloat)},
+				{Value: types.NewFloat64Datum(0.2), RetType: types.NewFieldType(mysql.TypeFloat)},
+			},
+		},
+	},
```

Comment:
Could you implement them in other_vec.go using template? So we can maintain all the cases of `IN` function in one way.